### ISSUE TITLE
Stream TypeAdapters

### ISSUE THREAD

**Issue by SalomonBrys** [SalomonBrys/Kotson#15]
This is an issue to track the integration of the Stream [TypeAdapter](https://google-gson.googlecode.com/svn/trunk/gson/docs/javadocs/index.html?overview-summary.html) API into Kotson.

The [JsonSerializer documentation](https://google-gson.googlecode.com/svn/trunk/gson/docs/javadocs/index.html?overview-summary.html) states:

> New applications should prefer TypeAdapter, whose streaming API is more efficient than this interface's tree API.

Kotson version `2.2.0-beta1` has been pushed to Maven Central and [exposes this API](https://github.com/SalomonBrys/Kotson/tree/2.2-beta#setting-custom-readers-and-writers).

Feel free to test it and report any concern or issue here ;)

**Comment by ralph-bergmann** [comment]
At first: It works :-)

But now my two points:
- it makes my GsonBuilder class really long :-(  (currently I have all my TypeAdapter classes in extra files)
- my TypeAdapters are surrounded with a TypeAdapterFactory[1] this makes it possible to inject the gson object to my TypeAdapters. But this is a edge case where I need the gson object in one of my TypeAdapter

[1] ... https://google-gson.googlecode.com/svn/trunk/gson/docs/javadocs/com/google/gson/TypeAdapterFactory.html

**Comment by SalomonBrys** [comment]
Regarding the idea to declare your `TypeAdapter`s in a different file, I've just pushed version `2.2.0-beta2` that enables this. I've updated [the documentation](https://github.com/SalomonBrys/Kotson/tree/2.2-beta#setting-custom-readers-and-writers) to explain how ;).

Regarding `TypeAdapterFactory`, since it's an interface that declares a generic function, there's currently no other way to use it other than implementing it on a regular object or class.

Hope this fits your needs :)
